File: python/hanoi_tower.py
def hanoi(n, frm, mid, to):
    if n == 1:
        return 1
    else:
        return hanoi(n-1, frm, to, mid) + 1 + hanoi(n-1, mid, frm, to)

moves = hanoi(32, 1, 2, 3)
print("%d moves" % moves)

File: go/hanoi_tower.go
package main

func hanoi(n uint, frm, mid, to byte) uint32 {
	if n == 1 {
		return 1
	}
	return hanoi(n-1, frm, to, mid) + 1 + hanoi(n-1, mid, frm, to)
}

func main() {
	moves := hanoi(32, 1, 2, 3)
	print(moves, " moves\n")
}
